internal/server/iigo.go

```diff
@@ -52,7 +53,7 @@ func (s *SOMASServer) runIIGOAllocations() error {
 		v.GameStateUpdate(newGameState)
 		if allocation <= s.gameState.CommonPool {
 			s.gameState.CommonPool -= allocation
```

Comment:
Why is this being reduced twice? Also, why is this not updating the server that the client now has more resources? This would cause a discrepancy to the client state in the start of the next round. 

Side note: The way common pool is changed will be modified when #124 is merged.

Reply:
Apologies, that is a bug, has now been patched, thanks